### ISSUE TITLE
Non-integer condition to generate with

### ISSUE THREAD

**Issue by psteinb** [atong01/conditional-flow-matching#163]
Dear all,

thanks again for the library. So far, I haven't found anything equivalent to exploit the conditional flow matching paradigm. 

To learn the library, I started from the example on [conditional generation](https://github.com/atong01/conditional-flow-matching/blob/main/examples/images/conditional_mnist.ipynb). I quickly stumbled upon the fact, that the enclosed `UnetModel` assumes very strongly that a condition (if provided) is a class label. In my case, the condition is a single floating point number or a `float` tensor. 

I sat down and adapted the `UnetModel` class accordingly. Everything seems to work there. I am currently moving towards testing an entire training loop. I am positing this here to hear, if 

a) people are interested in this?
b) this is implemented already and I just didn't see it?
c) it would be interesting to hear from @atong01 or @kilianFatras if you'd be happy to receive a PR when I am done? 

Some PRs look quite stale and I am unclear if you intend to maintain this library in the near future.

Best,
P

**Comment by atong01** [comment]
Hi @psteinb!

Thanks for your note! 

Curious what your use case is. Would be interested to see and happy to receive a PR. I never really looked too much at the UnetModel as we didn't use class conditioned in this work and I've moved away from image generation tasks. 

I guess I'm in maintenance mode for this project. Basically making sure it compiles and works but not working on new features at the moment.

**Comment by psteinb** [comment]
Ok, then give me a couple of days to do a minimal example. I'll share it here or point you to a topic branch in my fork. Thanks for replying to quickly.

**Comment by kilianFatras** [comment]
Hi all,

a) people are interested in this? I have no idea tbh...

b) this is implemented already and I just didn't see it? To the best of my
knowledge, it is not.

c) it would be interesting to hear from @atong01
<https://github.com/atong01> or @kilianFatras
<https://github.com/kilianFatras> if you'd be happy to receive a PR when I
am done?

We are always happy to receive contributions from the community and thank
you for using TorchCFM! However, I would like to express a concern here. I
think we can extend existing architectures but I am not open to change the
default behavior. After all this time, quite a lot of repo rely on
TorchCFM and we simply cannot break their dependencies out of nowhere. I am
also pretty sure a lot of people use torchcfm as a pedagogical tool to
understand how FM works and get quite good results on Cifar10 image
generation with our single or multi-gpu examples. Therefore, if the default
Unet architecture changes is extended, we have to ensure that the current
behavior remains exactly the same. It means that: i) all current tests pass
and ii) all reported results AND examples are reproducible with the new
architecture. This is also expected in good collaborative software
development (I am referring to SOLID principles - open/closed principle).

Sorry to be so demanding here, but now that torchcfm has been battle-tested
for such a long time, I do not wish it to break. I am happy to double check
and review the PR as well.

Best,
Kilian

Le mer. 7 mai 2025 à 11:36, Peter Steinbach ***@***.***> a
écrit :

**Comment by psteinb** [comment]
I took a stab at this, see #164 

The notebook [here](https://github.com/psteinb/conditional-flow-matching/blob/psteinb-nonclassification-labels/examples/images/conditional_mnist_noninteger.ipynb) demonstrates what is going on. The label can surely be expanded to more tricky objects. Haven't tested yet with real-world data or other benchmarks which allow conditioned generation based on a float or vector-of-floats label.
